Feat/Add notary notification for container contract

carpawell opened the pull request:
Closes #731.

Also: 
- merges `ContractProcessor` and `NotaryContractProcessor` interfaces;
- adds notary notification support for `delete` container operation;
- improves logging in notary listener;
- refactors binding processors and parsers/handlers.

Merge after #814.

alexvanin commented:
Check if container size announcements works correctly. I suppose they are not, because Storage node use container wrapper with `TryNotary()` option to invoke `putContainerSize` method. This method is not supposed to be invoked with notary service.

carpawell commented:
@alexvanin, yes, it seems that `putContainerSize` is the only one method that SN invokes and that does not work with notary, fixed that in https://github.com/nspcc-dev/neofs-node/pull/807/commits/a38dfd3d3c318811f71a783de661c998d7f670ea

also, please, take a look at https://github.com/nspcc-dev/neofs-node/pull/807/commits/704788e06df05f71f722c018cd71f939c326b40e, I added unexpected opcodes check to `Put` parser and reuse that approach in `setEACL` parser

also improved a little logging in https://github.com/nspcc-dev/neofs-node/pull/807/commits/8862d5e361402a3276b1e07440d31d24ed2cbb0d

alexvanin commented on pkg/morph/event/listener.go:
```diff
@@ -368,6 +368,11 @@ func (l listener) parseAndHandleNotary(nr *response.NotaryRequestEvent) {
 	}
 
+	log := l.log.With(
+		zap.String("contract_LE", notaryEvent.ScriptHash().StringLE()),
```
I suggest to assume that all string values are `LE`, because it is human readable format by default for Neo related things. Or you really want to emphasize encoding?

carpawell replied:
no, i do not want

just saw "script hash LE"

deleted "LE"
